src/resources/2021/CrystalBall.ts

```diff
@@ -1,38 +1,29 @@
 import {
   Location,
   Monster,
-  myTurncount,
   toLocation,
   toMonster,
+  availableAmount,
+  Item,
+  visitUrl,
 } from "kolmafia";
 import { get } from "../../property";
 
+export const orb = Item.get("miniature crystal ball");
+export function have(): boolean {
+  return availableAmount(orb) > 0;
+}
 const parsedProp = () =>
   get("crystalBallPredictions")
     .split("|")
     .map((element) => element.split(":") as [string, string, string])
     .map(
-      ([turncount, location, monster]) =>
-        [parseInt(turncount), toLocation(location), toMonster(monster)] as [
-          number,
-          Location,
-          Monster
-        ]
+      ([, location, monster]) =>
+        [toLocation(location), toMonster(monster)] as [Location, Monster]
     );
 
-export function currentPredictions(withFree = true): Map<Location, Monster> {
-  const predictions = parsedProp();
-  const freeCondition = (predictedTurns: number, turns: number) =>
-    predictedTurns === turns;
-  const nonFreeCondition = (predictedTurns: number, turns: number) =>
-    predictedTurns + 1 === turns;
-  return new Map(
-    predictions
-      .filter(
-        ([turncount]) =>
-          nonFreeCondition(turncount, myTurncount()) ||
-          (withFree && freeCondition(turncount, myTurncount()))
-      )
-      .map(([, location, monster]) => [location, monster])
-  );
+export function ponder(): Map<Location, Monster> {
```

Comment:
i think it is worth noting that once ponder was introduced i kind of thought this was the best way for libram to handle it so i am pretty overjoyed that we are finally just letting kol do it rather than trying to backend the comprehension of orb into libram. this is technically less efficient but i don't give the remotest shit, it works and is guaranteed to work regardless of any weird wanderer parsing or nonsense, thank you for doing this.